sources/api/early-boot-config/Cargo.toml

```diff
@@ -26,6 +27,10 @@ tokio = { version = "0.2", default-features = false, features = ["macros", "rt-t
 #tokio = { version = "0.3", default-features = false, features = ["macros", "rt-multi-thread"] }
 toml = "0.5"
 
+[target.'cfg(target_arch = "x86_64")'.dependencies]
```

Comment:
So if someone is doing a `cargo build` (so, separate from #1404 solutions), what happens here?  This dependency here would be skipped, but the code still references `vmw_backdoor::bla`, so I think it'd have compile errors about unknown references, which could be pretty confusing.  ARM developer machines are getting more common now, and vmware-dev is our first "local" variant, so I don't think this is "weird" enough to ignore.  Do we need to conditionalize the VmwareDataProvider on target_arch?

Reply:
With #1431 in, this is a lot less important.  I'm assuming we'd still have undefined references if `cargo build`ing this on a non-x86 host, so It'd be nice to do something to clean that up and make it clearer that it's not supported, but it's not as big an issue now since we'd expect the person to use `cargo make` to handle cross-compilation automatically.